Provide a compatibility alias __memcpy

For some reason, libelf uses the internal glibc alias __memcpy, which
doesn't exist in uClibc. Add a manual alias so that the build can
proceed with uClibc.

Based on the former patch by Thomas Petazzoni.

Signed-off-by: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>
Signed-off-by: Vicente Olivert Riera <Vincent.Riera@imgtec.com>

diff -rup a/libelf/libelf.h b/libelf/libelf.h
--- a/libelf/libelf.h	2014-08-27 10:25:17.000000000 +0100
+++ b/libelf/libelf.h	2014-11-07 15:13:08.743508221 +0000
@@ -34,6 +34,11 @@
 /* Get the ELF types.  */
 #include <elf.h>

+#ifndef _LIBC
+#ifndef __mempcpy
+#define __mempcpy mempcpy
+#endif
+#endif

 /* Known translation types.  */
 typedef enum
